perun-core/src/main/java/cz/metacentrum/perun/core/entry/MembersManagerEntry.java

```diff
@@ -1729,6 +1732,54 @@ public List<RichMember> getRichMembersByIds(PerunSession sess, List<Integer> ids
 		return membersManagerBl.filterOnlyAllowedAttributes(sess, richMembers);
 	}
 
+	@Override
+	public List<FailedCandidate> addMemberCandidates(PerunSession sess, Vo vo, List<MemberCandidate> candidates) {
+		Utils.checkPerunSession(sess);
+
+		List<FailedCandidate> failedCandidates = new ArrayList<>();
+
+		for (MemberCandidate candidate : candidates) {
+			try {
+				if (candidate.getRichUser() != null) {
+					Member member = this.createMember(sess, vo, candidate.getRichUser());
+					getPerunBl().getMembersManager().validateMemberAsync(sess, member);
```

Comment:
you can unify the validation calls and always call Bl layer (in the calls below instead of entry layer's validation method)